packages/interfaces/src/kobo/utils.ts

```diff
@@ -21,50 +22,62 @@ export const isDrought = (form: DisasterDtoType): form is DroughtDto => {
   return koboKeys[DROUGHT].disTyp in form;
 };
 
+export const isFloodArray = (
+  forms: FloodDto[] | DroughtDto[] | IncidentDto[],
+): forms is FloodDto[] => {
+  return koboKeys[FLOOD].disTyp in forms[0];
+};
+
+export const isDroughtArray = (
+  forms: FloodDto[] | DroughtDto[] | IncidentDto[],
+): forms is DroughtDto[] => {
+  return koboKeys[FLOOD].disTyp in forms[0];
+};
+
 export const formatCommonFields = (form: DisasterDtoType) => {
   if (isFlood(form)) {
     const keys = koboKeys[FLOOD];
 
     return {
-      province: form[keys.province],
-      district: form[keys.district],
-      commune: form[keys.commune],
-      disasterDate: form[keys.disasterDate],
-      disasterType: form[keys.disTyp],
-      reportName: form[keys.entryName],
-      phone: form[keys.phone],
-      entryDate: form[keys.entryDate],
-      id: form[keys.id],
+      [KoboCommonKeys.province]: form[keys.province],
+      [KoboCommonKeys.district]: form[keys.district],
+      [KoboCommonKeys.commune]: form[keys.commune],
+      [KoboCommonKeys.disasterDate]: form[keys.disasterDate],
+      [KoboCommonKeys.disTyp]: form[keys.disTyp],
+      [KoboCommonKeys.entryName]: form[keys.entryName],
+      [KoboCommonKeys.phone]: form[keys.phone],
+      [KoboCommonKeys.entryDate]: form[keys.entryDate],
+      [KoboCommonKeys.id]: form[keys.id].toString(),
```

Comment:
Again, these parts feel like they could be deduped between incident types

Reply:
I have type issues when I dedup. I am creating a ticket to handle that later